modular_zubbers/code/modules/status_indicators/status_indicator.dm

```diff
@@ -55,24 +57,18 @@ GLOBAL_LIST_INIT(potential_indicators, list(
 /datum/component/status_indicator/RegisterWithParent()
 	attached_mob = parent
 	// The Basics
-	RegisterSignal(parent, COMSIG_LIVING_DEATH, PROC_REF(cut_indicators_overlays))
 	RegisterSignal(parent, COMSIG_LIVING_LIFE, PROC_REF(status_indicator_evaluate))
 	// When things actually happen
 	RegisterSignal(parent, COMSIG_LIVING_STATUS_STUN, PROC_REF(status_indicator_evaluate))
 	RegisterSignal(parent, COMSIG_LIVING_STATUS_KNOCKDOWN, PROC_REF(status_indicator_evaluate))
 	RegisterSignal(parent, COMSIG_LIVING_STATUS_PARALYZE, PROC_REF(status_indicator_evaluate))
 	RegisterSignal(parent, COMSIG_LIVING_STATUS_IMMOBILIZE, PROC_REF(status_indicator_evaluate))
 	RegisterSignal(parent, COMSIG_LIVING_STATUS_UNCONSCIOUS, PROC_REF(status_indicator_evaluate))
-	RegisterSignal(parent, COMSIG_MOB_LOGIN, PROC_REF(apply_pref_on_login))
 
-/datum/component/status_indicator/proc/apply_pref_on_login()
-	var/atom/movable/screen/plane_master/rendering_plate/game_plate/local_status = locate() in attached_mob.client.screen
-	if(local_status)
-		. = attached_mob.client.prefs.read_preference(/datum/preference/toggle/enable_status_indicators)
-		local_status.alpha = (.) ? 255 : 0
+
 
 /datum/component/status_indicator/UnregisterFromParent()
-	QDEL_NULL(status_indicators)
+	QDEL_LIST(status_indicators)
```

Comment:
```suggestion
	QDEL_LIST_ASSOC_VAL(status_indicators)
```
This is an associative list where the keys are strings and the values are instances, the code for QDEL_LIST is `#define QDEL_LIST(L) if(L) { for(var/I in L) qdel(I); L.Cut(); }`, all that's going to do is try to delete a bunch of strings because doing a for loop on an assoc list loops by key, QDEL_LIST_ASSOC_VAL is `#define QDEL_LIST_ASSOC_VAL(L) if(L) { for(var/I in L) qdel(L[I]); L.Cut(); }` which loops through the list and deletes the value of the assoc entry, not the key. Also this needs to be in a Destroy() override because UnregisterFromParent doesn't always mean the component is deleting